src/components/verification-grid/verification-grid.ts

```diff
@@ -893,49 +866,161 @@ export class VerificationGridComponent extends AbstractComponent(LitElement) {
+    );
   }
 
-  private removeDecisionHighlight(selectedTiles: VerificationGridTileComponent[] = Array.from(this.gridTiles)): void {
-    for (const tile of selectedTiles) {
-      tile.decisionIndex = undefined;
+  // for verification tasks, the user will be adding one verification decision
+  // to each grid tile. Therefore, we can test that there is some sort of
+  // verifications applied to every tile
+  private hasOutstandingVerification(): boolean {
+    // we short circuit this function if there are no possible verification
+    // decisions
+    if (!this.hasVerificationTask()) {
+      return false;
     }
 
-    this.removeDecisionButtonHighlight();
+    const verificationTiles = this.gridTiles;
+    for (const tile of verificationTiles) {
+      const hasVerificationDecision = tile.model.verification !== undefined;
+
+      if (!hasVerificationDecision) {
+        return true;
+      }
+    }
+
+    return false;
   }
 
-  private touchedDecisionElements(): DecisionComponent[] {
-    const decisionElements = this.decisionElements;
-    if (!decisionElements) {
-      throw new Error("No decision elements found");
+  private outstandingTags(tile: VerificationGridTileComponent): Tag[] {
+    // TODO: there's probably a better way to do this than a n^2 + n operation
+    return this.classificationDecisionElements
+      .map((element: ClassificationComponent) => element.tag)
+      .filter((tagName) =>
+        !tile.model.classifications.some(
+          (classification) => classification.tag.text === tagName.text
+        )
+      );
+  }
+
+  // during a classification task, we want to ensure that every tile has a
+  // decision about every tag (not classification decision)
+  // we don't check against classification decisions because we want to support
+  // adding "negative" classifications against a tile
+  private hasOutstandingClassification(): boolean {
+    // because computing outstanding classification tasks is expensive, we can
+    // short circuit this function if there are no classification tasks
+    if (!this.hasClassificationTask()) {
+      return false;
+    }
+
+    // we use the tag to check if there are outstanding classifications
+    // (instead of the decision identifier) so that negative/positive decisions
+    // on the same tag are not counted as two separate decisions
+    // meaning that we only have to make a classification about each tag once
+    const requiredTags: Tag[] =
+      this.classificationDecisionElements
+        .map((element: ClassificationComponent) => element.tag);
+
+    const verificationTiles = this.gridTiles;
+    for (const tile of verificationTiles) {
+      const tileTags = tile.model.classifications.map((classification) => classification.tag.text);
+      const tileHasRequiredTags = requiredTags.every((tag) => tileTags.includes(tag.text));
+
+      if (!tileHasRequiredTags) {
+        return true;
+      }
+    }
+
+    return false;
+  }
```

Comment:
ok i have a gut feeling all of this could be reduced to about 5 lines of code. Need to have a think about this.

If we could ask each subject what decisions it has (e.g. `tile.subject.hasDecisions(allRequiredDecisions)` this is just a forr loop, no?

I also don't understand the `outstandingTags` function.